qiskit_experiments/framework/composite/composite_analysis.py

```diff
@@ -206,9 +207,24 @@ def _marginalized_component_data(self, composite_data: List[Dict]) -> List[List[
                         sub_data["counts"] = datum["counts"]
                 if "memory" in datum:
                     if composite_clbits is not None:
-                        sub_data["memory"] = (
-                            np.array(datum["memory"])[composite_clbits[i]]
-                        ).tolist()
+                        # level 2
+                        if isinstance(datum["memory"][0], str):
+                            num_cbits = 1 + max(
```

Comment:
It might be good to wrap this up in a `marginal_memory` function.

Reply:
After realizing that the code was inefficient (calling `format_counts_memory` more times than needed) I refactored it here: [871400e](https://github.com/Qiskit/qiskit-experiments/pull/801/commits/871400e7da59d7a70322f0452932d93be2a3c26a) where I also carved out part of the logic into a separate function.